sdk/core/azure_core/src/process/mod.rs

```diff
@@ -0,0 +1,51 @@
+use async_trait::async_trait;
+use std::io::Error;
+use std::{ffi::OsStr, process::Output};
+
+#[cfg(not(feature = "tokio_process"))]
+mod thread;
```

Comment:
You should never use mutually-exclusive features as called out in rust docs. The resolver does a union.

Reply:
There is no feature that's mutually exclusive with `tokio_process`: either it's present, or it's not. If there would be multiple features, from multiple backends, then there would be a problem indeed. But this isn't the case here.

Reply:
I see this matches what we do for `HttpClient`. While we could maybe keep this to make it easy for us to call - or even external callers, though I don't imagine much reason to - how it gets passed to credentials will have to change. I was playing around with some different options including a "static trait" for a runner, but the DX for that because of a bug/design issue in rustc isn't great.

So a slight adaptation of this + the options pattern in https://gist.github.com/heaths/ab062b63cac36bc1ef774a43c74dcf63 and this should work. Since you said your branch is locked down, I'll start a new one but add you as `Co-Authored-By`. Thanks for all your work here!

Reply:
> I see this matches what we do for `HttpClient`.

Yeah i've been inspired by that.

> Thanks for all your work here!

You are welcome. Great that I could have been of help.

I will keep this PR open until you make your pull request, but feel free to close it now if you prefer that.